### DISCUSSION TITLE
How do I debug or restart wings/panel?

### DISCUSSION THREAD

**Discussion by ElijahRus250** [pterodactyl/panel#2827] [General Community]
How to debug and find out where your issue is coming from and how to fix it.

First off we need to find out where your issue is coming from, wings or panel?

Lets find that out my typing `systemctl status wings` for your wings, normally it will show if there is an error here if not then its probably something to do with your panel or web server. 
 *if you are having a wings issue please skip down to the part about wings*

**PANEL ISSUES**
Panel Logs:
First off lets get your panel logs you can do so by typing the command below: 
`tail -n 100 /var/www/pterodactyl/storage/logs/laravel-$(date +%F).log | nc bin.ptdl.co 99`

Now this may look scary but trust me its not really that hard to understand. 
You are going to look for the errors, it should be quite easy to see where the errors are, then we are going to go to either the discord and search for past people having the same issue 
<img width="419" alt="Screen Shot 2020-12-08 at 11 22 17 PM" src="https://user-images.githubusercontent.com/64851720/101584684-3fa70280-39ac-11eb-9966-6269c007a38d.png">
If that does not work go to google and try some answers there if all else fails, ask in #help on discord.gg/pterodactyl

**WINGS ISSUES**
The wings are what connect both the panel and the servers together, if your wings are broken then we are going to need to debug. Sometimes it can be a very simple fix so lets try that one first. 
What we are going to be doing whilst running this command is we are going to reset the files wings and restart the wings all at the same time.
Run this command: `systemctl reset-failed wings && systemctl restart wings`

If this succeeds it should say no message in your console, if it does not work please continue on reading.

Another very simple way to attempt to fix your wings is copy the config file found in your node on your admin panel back into the correct spot on your server, This file should be placed in your daemon's root directory (usually `/etc/pterodactyl`) in a file called `config.yml`.
After this you will need to restart your wings again, run the command `systemctl restart wings`

If it is STILL not working correctly we are going to need to debug, next we are going to need to grab the wings logs, you can do so by running the command below: 

`tail -n 100 /var/log/pterodactyl/wings.log | nc bin.ptdl.co 99`

Aswell as run the command below:

Wings Diagnostics:
`wings diagnostics`

These logs will contain a lot of information, you are going to need to look for the errors *normally they are highlighted in red*. 

take those errors and search the pterodactyl discord for someone else having the same issue as chances are, you are not the first person to have that error. If you can't find an answer there you can try google but if that doesn't work, ask in #help 

I would also recommend you checking out this page https://pterodactyl.io/wings/ for more debugging tips!

**Comment by ElijahRus250** [comment]
did I forget any? maybe web servers logs, not sure

**Comment by DaneEveritt** [comment]
Can you please provide a little more depth to this discussion, explaining why you might need to be doing each thing rather than just providing random commands with no context. Remeber, people reading these discussion threads are likely looking for a specific answer, or have no idea what they're trying to find.

Random commands aren't going to help.

**Reply by ElijahRus250** [reply]
sure I will edit it now.

**Reply by ElijahRus250** [reply]
edited

**Comment by google-meet-meat** [comment]
Could you add to this? Sometimes in 0.7, the logs for the specific date haven't been created yet. Could you add that if a user gets the error `No such file or directory`, they should instead look in the log directory itself? `cd /var/www/pterodactyl/storage/logs`.

**Reply by ElijahRus250** [reply]
not sure how to add someone, if you want to give me a copy pasta all add to the bottom
